gtsam/base/OptionalJacobian.h

```diff
@@ -96,6 +97,20 @@ class OptionalJacobian {
     usurp(dynamic->data());
   }
 
+  /**
+   * @brief Constructor from an Eigen::Ref *value*. Will not usurp if dimension is wrong
+   * @note This is important so we don't overwrite someone else's memory!
+   */
+  OptionalJacobian(Eigen::Ref<Eigen::MatrixXd> dynamic_ref) :
+      map_(nullptr) {
+    if (dynamic_ref.rows() == Rows && dynamic_ref.cols() == Cols && !dynamic_ref.IsRowMajor) {
+      usurp(dynamic_ref.data());
+    } else {
+      // It's never a good idea to throw in the constructor
```

Comment:
Ok, then what?

Reply:
Since we don't allocate in this constructor it's probably fine though...